gcloud/resource_manager/client.py

```diff
@@ -0,0 +1,298 @@
+
+    @classmethod
+    def from_service_account_json(cls, json_credentials_path):
+        """Factory to retrieve JSON credentials while creating client.
+
+        Example::
+
+            >>> from gcloud.resource_manager import Client
+            >>> keyfile = '~/purple-spaceship-123-key.json'
+            >>> client = Client.from_service_account_json(keyfile)
+
+        :type json_credentials_path: string
+        :param json_credentials_path: The path to a private key file (this file
+                                      was given to you when you created the
+                                      service account). This file must contain
+                                      a JSON object with a private key and
+                                      other credentials information (downloaded
+                                      from the Google APIs console).
+
+        :rtype: :class:`gcloud.resource_manager.client.Client`
+        :returns: The client created with the retrieved JSON credentials.
+        """
+        credentials = get_for_service_account_json(json_credentials_path)
+        return cls(credentials=credentials)
+
+    @classmethod
+    def from_service_account_p12(cls, client_email, private_key_path):
+        """Factory to retrieve P12 credentials while creating client.
+
+        .. note::
+
+            Unless you have an explicit reason to use a PKCS12 key for your
+            service account, we recommend using a JSON key.
+
+        :type client_email: string
+        :param client_email: The e-mail attached to the service account.
+
+        :type private_key_path: string
+        :param private_key_path: The path to a private key file (this file was
+                                 given to you when you created the service
+                                 account). This file must be in P12 format.
+
+        :rtype: :class:`gcloud.resource_manager.client.Client`
+        :returns: The client created with the retrieved P12 credentials.
+        """
+        credentials = get_for_service_account_p12(client_email,
+                                                  private_key_path)
+        return cls(credentials=credentials)
+
+    def list_projects(self, filter_params=None, page_size=None,
+                      page_token=None):
+        """List the projects visible to this client.
+
+        Example::
+
+            >>> from gcloud import resource_manager
+            >>> client = resource_manager.Client()
+            >>> for project in client.list_projects():
+            ...     print project.project_id
+
+        List all projects with label ``'environment'`` set to ``'prod'``
+        (filtering by labels)::
+
+            >>> from gcloud import resource_manager
+            >>> client = resource_manager.Client()
+            >>> env_filter = {'labels.environment': 'prod'}
+            >>> for project in client.list_projects(env_filter):
+            ...     print project.project_id
```

Comment:
I think examples belong in "usage" docs, where conveying the gestalt is crucial, but not API docs, where they get in the way of the clarity needed specifying method / function contracts.